### PR TITLE
MAINT: interpolate: remove duplicated `splint` from `_fitpack`

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue

A part of #16729
This PR is another option to #16871

#### What does this implement/fix?

As reported in #16729, _fitpack and dfitpack have some dupricated interfaces of FITPACK.
This PR focuses on `splint` and remove the _fitpack interface.

### CODE REVIEW

File: scipy/interpolate/_bsplines.py [hunk 1]
```diff
@@ -657,8 +656,7 @@ def integrate(self, a, b, extrapolate=None):
             if self.c.ndim == 1:
                 # Fast path: use FITPACK's routine
                 # (cf _fitpack_impl.splint).
-                t, c, k = self.tck
-                integral, wrk = _dierckx._splint(t, c, k, a, b)
```
Comment (rgommers) [inline scipy/interpolate/_bsplines.py:661]: This is a nice to have change

File: scipy/interpolate/_fitpack_impl.py [hunk 2]
```diff
@@ -654,7 +654,13 @@ def splint(a, b, tck, full_output=0):
...
                         splint(a, b, [t, c, k]), c))
     else:
-        aint, wrk = _fitpack._splint(t, c, k, a, b)
+        if t.size != c.size:
+            if c.size >= t.size - k - 1:
+                # If t and c are not the same size, c is padded with zeros.
+                c = np.pad(c, (0, t.size - c.size))
+            else:
+                raise ValueError("t and c are invalid length.")
+        aint, wrk = dfitpack.splint(t, c, k, a, b)
```
Comment (rgommers) [inline scipy/interpolate/_fitpack_impl.py:663]: This is a downside, quite a bit of extra code.

File: scipy/interpolate/tests/test_bsplines.py [hunk 3]
```diff
@@ -887,6 +885,9 @@ def test_splint(self):
         assert_allclose(integr,
                         splint(0, 1, b), atol=1e-14)
 
+        with assert_raises(ValueError, match="t and c are invalid length."):
+            splint(0, 1, (np.ones(10), np.ones(5), 3))
```
Comment (rgommers) [inline scipy/interpolate/tests/test_bsplines.py:889]: Same issue I guess as the change in `_fitpack_impl.py`.

**Review by rgommers — COMMENTED** [review]
It's a toss-up which approach is better it looks like. If dealing with "t and c are not the same size" could be dealt with as a constraint in the `.pyf` file, then this PR would be a clear win.

I'll defer to @ev-br's opinion here though.

**Comment by ev-br** [comment]
gh-17038 gives a third alternative, following @rgommers 's comment above. 

As to which one to remove from, `_fitpack` or `dfitpack`, I'm mostly following Pauli's original suggestion, https://github.com/scipy/scipy/issues/1408#issuecomment-24493859, *Get rid of _fitpack in favor of dfitpack*, and the 
*Try to replace _fitpack interface with f2py-generated version* comment , https://github.com/scipy/scipy/blob/main/scipy/interpolate/_fitpack_impl.py#L33, which `$ git blame -M -C -C` attributes to Travis in his 2006 commit e315609f135.

File: scipy/interpolate/_fitpack_impl.py (hunk 2 above)
Comment (ev-br) [reply scipy/interpolate/_fitpack_impl.py:663]: It rather embarassingly took me a while to realize that Fortran does not really allow variable-sized arrays (unless `allocatable`), so the f2py declaration `real*8 dimension(n)` does not allow for arrays of size less than `n`. So every array needs to be accompanied by its integer size variable, and an f2py check can track if `len(c) >= len(t) - k - 1`, as in gh-17038.

**Comment by ev-br** [comment]
The contents of this PR is merged as a part of gh-17038.  Thanks again @AtsushiSakai

**Pull request closed without merging**